util/types/json/compare.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The ql Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSES/QL-LICENSE file.
+
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"bytes"
+	"reflect"
+	"strings"
+)
+
+// compareFloat64 returns an integer comparing the float64 x to y,
+// allowing precision loss.
+func compareFloat64PrecisionLoss(x, y float64) int {
+	var EPSILON = 0.00000001
+	if x-y < EPSILON && y-x < EPSILON {
+		return 0
+	} else if x-y < 0 {
+		return -1
+	}
+	return 1
+}
+
+// jsonTypePrecedences is for compare two json.
+// See: https://dev.mysql.com/doc/refman/5.7/en/json.html#json-comparison
+var jsonTypePrecedences = map[string]int{
+	"BLOB":     -1,
```

Comment:
Yes, it's just from MySQL 5.7 document. Maybe in future we will add those types into JSON, so I prefer keep it. what do you think?